### ISSUE TITLE
The `punycode` module is deprecated

### ISSUE THREAD

**Issue by sergiojtsm** [makenotion/notion-sdk-js#486]
The `punycode` module is deprecated, please fix it as soon as possible

**Comment by leafstrat** [comment]
Nothing to do with notion.

**Comment by adibkadir** [comment]
I'm getting a warning when using notion SDK
`(node:30366) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)`

**Comment by adibkadir** [comment]
Looks like an issue with v20+ and older version of pnpm on my end. update of pnpm fixed it.

**Comment by mazerte** [comment]
Yes the error pops on Node v20+ but it's due to `whatwg-url` dependency imported by `node-fetch`. The v3.x of `node-fetch` fix the problem. I will be good if notion can update this dependency.
